### DISCUSSION TITLE
flopy.mf6.MFSimulation.load ERROR

### DISCUSSION THREAD

**Discussion by ArtCoff** [modflowpy/flopy#2279] [Q&A] [answered]
I create a modflow6 model with flopy and saved it locally.
![image](https://github.com/user-attachments/assets/2186c80c-f03c-4ea9-b89a-1a925f82ff44)
![image](https://github.com/user-attachments/assets/6d2f19e3-dd80-433a-b5f6-db6549776ff9)
The local directory is  "D://mf6_cache"
Then I loaded it using theflopy.mf6.MFSimulation.load method and got an error.
![image](https://github.com/user-attachments/assets/e300493d-992e-4e11-9ea4-dac5736762b1)

![image](https://github.com/user-attachments/assets/ccb05049-907c-44bd-9a80-98adcc31f381)
I don't understand why.

**Comment by dbrakenhoff** [comment]
Hi @GroundwaterSimulationOptimization,

The MF6Simulation NAM-file is called "mfsim.nam":

```python
sim = flopy.mf6.MFSimulation.load("mfsim.nam", sim_ws="path)
```

**Reply by ArtCoff** [reply]
![image](https://github.com/user-attachments/assets/2d45f49e-16c8-45c1-adc5-d4dcb041a304)
this did not work

**Reply by ArtCoff** [reply]
I don't understand what the parameter sim_name means, does it need a mock name

**Reply by dbrakenhoff** [reply, marked as answer]
Ah, the `sim_name` isn't actually used to look for the Simulation NAM-file apparently which is always `mfsim.nam`. So good question, not sure what it's used for other than some internal naming.

But something is going wrong in loading the Storage package apparently. It seems to think the package data is empty... What version of flopy/mf6 are you using? Are you able to share the model here? If so, I, or one of the actual flopy gurus, can take a look. 

If possible, you can try skipping the STO package for now using the load_only kwarg and passing a list of packages it should load, e.g. `flopy.mf6.MFSimulation.load(sim_ws=path, load_only=["DIS", "IC", "OC", ...])`. 

>  In the past I used modflow2005's load method and felt it was very clear, however mf6's load method is not well presented. I feel like he's asking me to do the compute on the source model file.

I'm not sure what you mean by this last comment. Could you elaborate? I agree the sim_name not amounting to to much is confusing though. But it should usually be sufficient to call `flopy.mf6.MFSimulation.load(sim_ws=<path to simulation>)` and that will usually load your simulation.

**Reply by wpbonelli** [reply]
> > In the past I used modflow2005's load method and felt it was very clear, however mf6's load method is not well presented.

> I agree the sim_name not amounting to to much is confusing though.

Noted &mdash; feedback like this is valuable and much appreciated as we prototype the next flopy.

**Reply by ArtCoff** [reply]
Thank you very much for your reply, after re-modifying the sto package it finally got it to work!

**Reply by ArtCoff** [reply]
On another point, here's a code I used to use a lot to load modflow models from local, you can see that my local file is passed into the .nam file as a path, and model_ws is the directory used for the run, which ensures that the output file for the run is separate from the model source file, but the way mf6 introduces it is directly on my original model file output, overwriting my original files.
mf = flopy.modflow.Modflow.load(path, model_ws=sim_ws, exe_name="mf2005", version="mf2005" )
My work requires a lot of repeated calls to models and constant modifications to them, and it's important to ensure the independence of the original files, which will hopefully be optimized in subsequent releases!

**Reply by dbrakenhoff** [reply]
I'm not sure if this answers your question, but you can load a simulation and then modify the workspace directory to avoid overwriting the simulation you just loaded. 

```python
sim = flopy.mf6.MFSimulation.load(sim_ws=path)
sim.set_sim_path(new_dir)
```
Additionally, you can also change the package names, which would rename the packages so you don't overwrite the original model files:

```python
sim = flopy.mf6.MFSimulation.load(sim_ws=path)
sim.rename_all_package("new_name")
```

**Reply by ArtCoff** [reply]
Wonderful！

**Comment by ArtCoff** [comment]
In the past I used modflow2005's load method and felt it was very clear, however mf6's load method is not well presented.I feel like he's asking me to do the compute on the source model file.
![image](https://github.com/user-attachments/assets/a66b4c3e-aeec-4bc6-b27e-7b12e26f9bfe)
